pymc_experimental/utils/model_fgraph.py

```diff
@@ -182,10 +185,28 @@ def fgraph_from_model(
 
     memo = {}
 
-    # Replace RNG nodes so that seeding does not interfere with old model
-    for rng in find_rng_nodes(model_vars):
-        new_rng = shared(rng.get_value(borrow=False))
-        memo[rng] = new_rng
+    # Replace the following shared variables in the model:
+    # 1. RNGs
+    # 2. MutableData (could increase memory usage significantly)
```

Comment:
We'll have to wait and get some feedback. If people complain about this, we can review the memory sharing.